### ISSUE TITLE
cacti upgrade issue

### ISSUE THREAD

**Issue by my22am** [KnoAll/cacti-template#176]
hi there, I believe the new version of Cacti package is moved? When I try to update it shows the following error msg:

::: Fri_09:01: Beginning Cacti upgrade...

gzip: stdin: unexpected end of file
tar: Unexpected EOF in archive
tar: Unexpected EOF in archive
tar: Error is not recoverable: exiting now
!!! ERROR: Fri_09:01: Cacti unpack error cannot install, exiting...

Thanks for looking into this.

**Comment by KnoAll** [comment]
Sorry for the delay, did not see this until now. Can you describe at what point you're seeing this? Is that from the upgrade script?

**Comment by my22am** [comment]
I believe its somewhere on "upgrade-cacti.sh" (line 406)

wget -q https://github.com/Cacti/cacti/archive/release/$prod_version.tar.gz

I think this path/URL is no longer valid.

by the way, there is also a new version released recently. :)

Thanks for checking!

**Comment by KnoAll** [comment]
I’m not sure why this is happening. The file appears to download for me. It’s a weird github link so maybe that is too complicated. I may switch back to the original release repository on cacti.net.

Also for the new v1.2.28, an error was found that prevents graphs from being automatically refreshed in the UI (data still gathers) so I have not pushed the update yet.

**Comment by my22am** [comment]
hi Kevin,

Good news, for some reason it's working again! I was able to upgrade!

However, I ran into issue or restoring the data...
I took backup from Cacti Centos 7 ova (upgraded the cacti to 1.2.27 previously)
I ran the "--restore-data" option and it report restored successfully but I could not see my old data...



*** WARNING: Fri_13:56: Beginning Cacti data restore process...
::: Fri_13:56: Valid Cacti install found...
::: Fri_13:56: The following Cacti Backup archives were found; select one:
1) backup_cacti-20241011.tar.gz
Use number to select a file or 'stop' to cancel: 1
::: Fri_13:56: Unpacking backup...
:::
Cacti v1.2.27 found, is that what you want to restore? [y/N] y
Are you REALLY sure you want to restore? This will destructively overwrite the existing Cacti installation and is irreversible. [y/N] y
::: Fri_13:57: Restoring Cacti v1.2.27 from backup...
Do you want to backup existing Cacti install before restoring over top? [y/N] n
*** WARNING: Fri_13:57: Skipping backup of existing Cacti.
::: Fri_13:57: Restoring Cacti DB...
Enter password:
:::
::: Fri_14:00: Restoring RRA data...
:::
::: Fri_14:00: Restoring Config...
:::
::: Fri_14:00: Cleaning up source files...
:::
::: Fri_14:00: Cacti v1.2.27 was successfully restored. You may now proceed to the web interface.

Not sure if I miss anything?  the cacti data was not restored. :-/

also there is one step it's asking for password right after the DB is restored, could that be the db password?  (which I assume it's "cacti")

::: Fri_13:57: Restoring Cacti DB...
Enter password:   (( here -- which password should I enter?))


Again thanks for the help!

**Comment by KnoAll** [comment]
Did the database restore and, but the graphs are empty?

**Comment by my22am** [comment]
it does seems like it

i got a message saying

**Fri_14:00: Cacti v1.2.27 was successfully restored. You may now proceed to the web interface.**

is there a way for me to check DB side?

**Comment by KnoAll** [comment]
Do all your devices show up in the console? Is data gathering again?
It may be that the rra folder did not restore across.

The DB manages what devices and all the parameters, the graph data is stored in RRA flat file.

**Comment by my22am** [comment]
i think RRA Folders are restored.  Its the DB portion that I am not sure of.

Maybe I missed this part?  

::: Fri_13:57: Restoring Cacti DB...
Enter password: (( here -- which password should I enter?))

**Comment by my22am** [comment]
and I dont see any of my devices at all too so I think its mostly on DB restore side...

**Comment by my22am** [comment]
Kevin,  

I think i figured out where my issue was.

I did a yum update before I ran cacti update script which i believe broke the DB.

Here is what I did.

1. download AlmaLinux ova from your site
2. Import ova
3. once boot up the VM, execute "./cacti-upgrade.sh" which would update a bunch of things.(php/db...etc)
4. Execute yum update  -y
5. now do a cacti import and it works!

Once again, Appreciated for the help! no more issues!
